pkg/route/api/types.go

```diff
@@ -133,20 +133,31 @@ type TLSConfig struct {
 	// termination this file should be provided in order to have routers use it for health checks on the secure connection
 	DestinationCACertificate string
 
+        // DEPRECATED: Keep this for backward compatibility
 	// InsecureEdgeTerminationPolicy indicates the desired behavior for
 	// insecure connections to an edge-terminated route:
 	//   disable, allow or redirect
 	InsecureEdgeTerminationPolicy InsecureEdgeTerminationPolicyType
+
+	// InsecureTerminationPolicy indicates the desired behavior for
+	// insecure connections to one of the tls (edge,passthrough,reencrypt) route:
+	//   disable, allow or redirect
+	InsecureTerminationPolicy InsecureTerminationPolicyType
```

Comment:
Why would we not just add "Redirect" to "InsecureEdgeTerminationPolicy"?  I don't see the value in adding the new external field - we can just rename it in a future version.  Broadening EdgeTerminationPolicy to apply to all termination types doesn't seem to have a disadvantage I can see.  Examples?